src/codeu/chat/ServerMain.java

```diff
@@ -95,10 +116,66 @@ private static void runServer(Uuid id,
                         new NoOpRelay() :
                         new RemoteRelay(relaySource);
 
-    final Server server = new Server(id, secret, relay);
+    //name of persistentLog file              
+    String persistentFileName = persistentPath + "//persistentLog.txt";
+    
+    final Server server = new Server(id, secret, relay, persistentFileName);
 
     LOG.info("Created server.");
-
+    
+    
+    
+    //location where persistentLog will be written to
```

Comment:
The below logic should be implemented in a separate method, in the very least, if not in a separate class. Hint: util